apps/libsignpost/signpost_api.h

```diff
@@ -261,23 +265,39 @@ int signpost_processing_reply(uint8_t src_addr, uint8_t message_type, uint8_t* r
 /* ENERGY API                                                             */
 /**************************************************************************/
 
+//message types
 enum energy_message_type {
     EnergyQueryMessage = 0,
     EnergyLevelWarning24hMessage = 1,
     EnergyLevelCritical24hMessage = 2,
     EnergyCurrentWarning60sMessage = 3,
+    EnergyReportMessage = 4,
+    EnergyDutyCycleMessage = 5,
 };
 
+//information sent to a module from the controller
 typedef struct __attribute__((packed)) energy_information {
-    uint32_t    energy_limit_24h_mJ;
-    uint32_t    energy_used_24h_mJ;
-    uint16_t    current_limit_60s_mA;
-    uint16_t    current_average_60s_mA;
+    uint32_t    energy_limit_mWh;
+    uint16_t    average_power_mW;
```

Comment:
The module uses this by dividing limit by the their power and seeing if it is above or below some target lifetime. The energy limit is just valid. There is no timeout. It decreases if you use energy, and it increases if the signpost successfully harvests energy. We could do some windowed update but it makes everyone's logic more complex. This is simple and it works.